module/VuFind/src/VuFind/Content/PageLocator.php

```diff
@@ -84,35 +84,102 @@ public function __construct($themeInfo, $language, $defaultLanguage)
         $this->defaultLanguage  = $defaultLanguage;
     }
 
+    /**
+     * Generate a template from a file search pattern. Examples:
+     * - %pathPrefix%/%pageName%{_%language%} => content/help_en
+     * - %pathPrefix%/%language%/%pageName% => HelpTranslations/en/search
+     *
+     * @param string $pathPrefix Subdirectory where the template should be located
+     * @param string $pageName   Page name
+     * @param string $pattern    Filesystem pattern
+     * @param string $language   Language
+     *
+     * @return string
+     */
+    protected function generateTemplateFromPattern(
+        $pathPrefix, $pageName, $pattern, $language=null
+    ) {
+        $template = $pattern;
+        $template = str_replace('%pathPrefix%', $pathPrefix, $template);
+        $template = str_replace('%pageName%', $pageName, $template);
+        $languagePatternExtended = '"\\{(.*)(%language%)(.*)\\}"';
```

Comment:
%language% doesn't need to be captured, so you can leave the surrounding () out and use \\2 in the replacement.